### PR TITLE
[Doc] Updated tpu-installation.rst with more details

### PR DESCRIPTION
**Pull request by mikegre-google**
Added more information about TPUs, including links to doc on TPU topology, TPU pricing, TPU runtime version, adding a persistent disk, and the different TPU  versions

FIX N/A

### CODE REVIEW

**Review by robertgshaw2-redhat — APPROVED** [review]

File: docs/source/getting_started/tpu-installation.rst [hunk 1]
```diff
@@ -1,35 +1,167 @@
...
+Create a TPU v5e with 4 TPU chips:
+
+.. code-block:: console
+
+    gcloud alpha compute tpus queued-resources create QUEUED_RESOURCE_ID \
+    --node-id TPU_NAME \
+    --project PROJECT_ID \
+    --zone ZONE \
+    --accelerator-type ACCELERATOR_TYPE \
+    --runtime-version v2-alpha-tpuv6e \
```
Comment (bvrockwell) [inline docs/source/getting_started/tpu-installation.rst:68]: runtime-version would need to be changed if this is intended for v5e i believe.

**Review by bvrockwell — APPROVED** [review]
Thanks @mikegre-google (just one minor fix), but otherwise... looks awesome!

**Comment by robertgshaw2-neuralmagic** [comment]
FYI @mikegre-google, the instructions above give the following error when running `vllm serve Qwen/Qwen2-1.5B-Instruct`

```bash
(venv) robertgshaw@t1v-n-afb619fd-w-0:~/vllm$ vllm serve Qwen/Qwen2-1.5B-Instruct
Traceback (most recent call last):
  File "/home/robertgshaw/vllm/venv/bin/vllm", line 33, in <module>
    sys.exit(load_entry_point('vllm', 'console_scripts', 'vllm')())
  File "/home/robertgshaw/vllm/venv/bin/vllm", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 171, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/robertgshaw/vllm/vllm/__init__.py", line 3, in <module>
    from vllm.engine.arg_utils import AsyncEngineArgs, EngineArgs
  File "/home/robertgshaw/vllm/vllm/engine/arg_utils.py", line 11, in <module>
    from vllm.config import (CacheConfig, ConfigFormat, DecodingConfig,
  File "/home/robertgshaw/vllm/vllm/config.py", line 12, in <module>
    from vllm.model_executor.layers.quantization import QUANTIZATION_METHODS
  File "/home/robertgshaw/vllm/vllm/model_executor/layers/quantization/__init__.py", line 5, in <module>
    from vllm.model_executor.layers.quantization.awq_marlin import AWQMarlinConfig
  File "/home/robertgshaw/vllm/vllm/model_executor/layers/quantization/awq_marlin.py", line 6, in <module>
    import vllm.model_executor.layers.fused_moe  # noqa
  File "/home/robertgshaw/vllm/vllm/model_executor/layers/fused_moe/__init__.py", line 4, in <module>
    from vllm.model_executor.layers.fused_moe.layer import (
  File "/home/robertgshaw/vllm/vllm/model_executor/layers/fused_moe/layer.py", line 22, in <module>
    from .moe_pallas import fused_moe as fused_moe_pallas
  File "/home/robertgshaw/vllm/vllm/model_executor/layers/fused_moe/moe_pallas.py", line 3, in <module>
    from torch_xla.experimental.custom_kernel import _histogram
  File "/home/robertgshaw/vllm/venv/lib/python3.10/site-packages/torch_xla/__init__.py", line 20, in <module>
    import _XLAC
ImportError: /home/robertgshaw/vllm/venv/lib/python3.10/site-packages/_XLAC.cpython-310-x86_64-linux-gnu.so: undefined symbol: _ZN5torch4lazy13MetricFnValueB5cxx11Ed
```

**Comment by robertgshaw2-neuralmagic** [comment]
note that the old installation instructions seem to work properly

**Comment by bvrockwell** [comment]
@robertgshaw2-neuralmagic we'll update with the following (unpinned torch_xla with a bug at head looks like:

will update with @mikegre-google 

```
pip3 install --pre torch==2.6.0.dev20241028+cpu torchvision==0.20.0.dev20241028+cpu \
    --index-url https://download.pytorch.org/whl/nightly/cpu
pip install "torch_xla[tpu] @ https://storage.googleapis.com/pytorch-xla-releases/wheels/tpuvm/torch_xla-2.6.0.dev20241028-cp310-cp310-linux_x86_64.whl" -f https://storage.googleapis.com/libtpu-releases/index.html
pip install torch_xla[pallas] -f https://storage.googleapis.com/jax-releases/jax_nightly_releases.html -f https://storage.googleapis.com/jax-releases/jaxlib_nightly_releases.html'

```

**Pull request merged**